packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -301,6 +308,8 @@ async function generateCacheEntryImpl(
     }
   )
 
+  // Track the timestamp when we started copmuting the result.
+  const startTime = performance.timeOrigin + performance.now()
```

Comment:
We track the time before we started computing anything so that we know that any changes after this timestamp might have changed this entry.